pkg/threescale/backend/backend.go

```diff
@@ -446,13 +454,13 @@ func (a *Application) calculateDeltas() api.Metrics {
 // addDeltasToRemoteState modifies the remote state with the provided deltas
 func (a *Application) addDeltasToRemoteState(deltas api.Metrics) *Application {
 	for metric, value := range deltas {
-		periods, ok := a.RemoteState[metric]
+		counters, ok := a.RemoteState[metric]
 		if !ok {
 			continue
 		}
 
-		for period, _ := range periods {
-			a.RemoteState[metric][period].current += value
+		for i, _ := range counters {
+			counters[i].CurrentValue += value
```

Comment:
idea: perhaps the code in `localReport` could use a part of `addDeltasToRemoteState` (a more generalised extracted func) with this logic about adding a number of hits to a set of metric counters?